pages/index.vue

```diff
@@ -152,37 +162,38 @@ function onMarkerReset() {
   leafletMapRef.value?.resetSelectedMarker();
 }
 
-watch(feature, async () => {
-  loading.value = true;
-  fetchedData.value = null;
-  seriesData.value = [];
-  isDataSeries.value = false;
-  selectedItem.value = null;
-  selectedIndex.value = 0;
-  if (feature.value) {
-    try {
-      const response = await $fetch(
-        `/api/fetchOpenData?feature=${encodeURIComponent(feature.value)}`,
-      );
-
-      const featureCollections = response as GeoJSON.FeatureCollection[];
-      isDataSeries.value = featureCollections.length > 1;
-
-      if (isDataSeries.value) {
-        seriesData.value = featureCollections;
-        dateOptions = getDateOptions(seriesData.value);
-        dateGroup = getDatesGroups(seriesData.value);
-        selectedIndex.value = dateOptions.length - 1;
-        selectedDate.value = dateOptions[selectedIndex.value];
+watch(feature, async (newval) => {
+  if (newval) {
+    loading.value = true;
+    fetchedData.value = null;
+    seriesData.value = [];
+    isDataSeries.value = false;
+    selectedItem.value = null;
+    selectedIndex.value = 0;
+    if (feature.value) {
+      try {
+        const response = await $fetch(
+          `/api/fetchOpenData?feature=${encodeURIComponent(feature.value)}`,
+        );
+        const featureCollections = response as GeoJSON.FeatureCollection[];
+        isDataSeries.value = featureCollections.length > 1;
+        showDetailButton.value = true;
```

Comment:
@alifarooq93 It is never set back to false so we don't need the variable as we can determine the flag transiently if a feature is loaded or not